### PR TITLE
HUD: Update selection mesh every frame

### LINKED ISSUE #3873: Incorrectly selected node after node change with node highlighting
screenshot_20160318_000806

To reproduce: place a door, and try to open it and close it multiple times until the bug is triggered. I can only reproduce it with node highlighting. Also, moving the mouse after triggering the bug can fix it, so you should only click the buttons to open and close the door after you selected it (a touchpad is useful for that).

### LINKED ISSUE #11904: Minetest doesn't update entity selectionbox while selected
##### Minetest version

```
5.4.1
```

##### Summary
If the `selectionbox` is changed using `set_properties` while the entity is being pointed at, the change will only be reflected in the displayed selectionbox if the pointed thing is changed. I wonder whether the same applies to nodes as well.

### PR DESCRIPTION
**Pull request by appgurueu**
Fixes #11904, most likely also fixes #3873

Minetest incorrectly made the assumption that the selectionbox doesn't change when the pointed thing doesn't change, which doesn't hold in the case of entities (which may have their properties changed) and nodes (which may have their neighborhood changed).

This PR fixes this by always updating the selection mesh. The performance impact is presumably negligible; worst-case performance (pointed thing constantly changing) hasn't changed at all. Only updating the selection mesh when it has changed would require way more code complexity and probably be less efficient.

### CODE REVIEW

**Comment by rubenwardy** [comment]
Could this also fix #3873 ?

**Comment by appgurueu** [comment]
> Could this also fix #3873 ?

Unless `updateSelectionMesh` is broken it definitely should.

File: src/client/game.cpp [hunk 1]
```diff
@@ -3093,10 +3093,10 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud)
 			!runData.btn_down_for_dig,
 			camera_offset);
 
-	if (pointed != runData.pointed_old) {
+	if (pointed != runData.pointed_old)
 		infostream << "Pointing at " << pointed.dump() << std::endl;
-		hud->updateSelectionMesh(camera_offset);
-	}
+
+	hud->updateSelectionMesh(camera_offset);
```
Comment (sfan5) [inline src/client/game.cpp:3099]:
```suggestion
	// Note that updating the selection mesh every frame is not particularly efficient,
	// but the halo rendering code is already inefficient which "cancels" this out.
	hud->updateSelectionMesh(camera_offset);
```
or similar

Comment (appgurueu) [reply src/client/game.cpp:3099]: what about "so there's no point in optimizing it here"?

Comment (sfan5) [reply src/client/game.cpp:3099]: thats ok too

**Review by sfan5 — APPROVED** [review]

**Review by SmallJoker — APPROVED** [review]
No noticeable "Draw scene" or "drawtime" increase. Works as expected.

**Pull request merged**